pico/tools/src/main/java/io/helidon/pico/tools/Item.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.tools;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+
+import io.helidon.builder.Builder;
+import io.helidon.builder.Singular;
+
+/**
+ * These are the individual items that compose the {@link ModuleInfoDescriptor} builder.
+ * <p>
+ * Note that generally speaking the majority of use cases can use methods from {@link ModuleInfoDescriptor}
+ * instead of needing to use anything provided here.
+ */
+@Builder
+public interface Item {
```

Comment:
The name seems to be very general. If this is related to module info, maybe it should be `ModuleInfoItem`?

Reply:
done; next commit